flink-runtime/src/main/java/org/apache/flink/runtime/state/PartitionableListState.java

```diff
@@ -45,23 +47,25 @@
     private final ArrayListSerializer<S> internalListCopySerializer;
 
     PartitionableListState(RegisteredOperatorStateBackendMetaInfo<S> stateMetaInfo) {
-        this(stateMetaInfo, new ArrayList<S>());
+        this(stateMetaInfo, new ArrayList<>(), stateMetaInfo.getPartitionStateSerializer());
+    }
+
+    // constructor for deep copy.
+    private PartitionableListState(PartitionableListState<S> toCopy) {
+        this(
+                toCopy.stateMetaInfo.deepCopy(),
+                toCopy.internalListCopySerializer.copy(toCopy.internalList),
+                toCopy.stateMetaInfo.getPartitionStateSerializer().duplicate());
```

Comment:
Looking at the `stateMetaInfo.deepCopy`, I see that `RegisteredOperatorStateBackendMetaInfo` already deep-copies the serializer. And then this copy is used in constructor currently in `master`.

The added test passes with the current (`master`) production code.

So it seems to me the change is not necessary.

Reply:
You are right, I will close the original ticket. Thanks for the review!